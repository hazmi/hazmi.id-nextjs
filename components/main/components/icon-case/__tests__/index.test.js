import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import IconCase from '../index'

describe('With Enzyme', () => {
  it('Show the correct default svg', () => {
    const wrapper = shallow(<IconCase />)
    expect(wrapper.find('path').prop('fill')).toEqual('#444')
  })
  it('Show the correct custom color', () => {
    const wrapper = shallow(<IconCase fill="red" />)
    expect(wrapper.find('path').prop('fill')).toEqual('red')
  })
})

describe('With Snapshot Testing', () => {
  it('Show the correct default svg', () => {
    const component = renderer.create(<IconCase />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Show the correct custom color', () => {
    const component = renderer.create(<IconCase fill="red" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})