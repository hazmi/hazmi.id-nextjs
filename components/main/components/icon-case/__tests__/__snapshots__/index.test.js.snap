// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Show the correct custom color 1`] = `
<svg
  fill="none"
  height={24}
  width={24}
>
  <path
    clipRule="evenodd"
    d="M10 3h4a2 2 0 0 1 2 2v1h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h4V5a2 2 0 0 1 2-2zM4 8h16v5H4V8zm0 11v-4h7v1h2v-1h7v4H4zM14 5v1h-4V5h4z"
    fill="red"
    fillRule="evenodd"
  />
</svg>
`;

exports[`With Snapshot Testing Show the correct default svg 1`] = `
<svg
  fill="none"
  height={24}
  width={24}
>
  <path
    clipRule="evenodd"
    d="M10 3h4a2 2 0 0 1 2 2v1h4a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h4V5a2 2 0 0 1 2-2zM4 8h16v5H4V8zm0 11v-4h7v1h2v-1h7v4H4zM14 5v1h-4V5h4z"
    fill="#444"
    fillRule="evenodd"
  />
</svg>
`;
